docs/resources/samples/filter-table-get-visible-range.md

```diff
@@ -0,0 +1,127 @@
+---
+title: 'Filter Excel table and get visible range'
+description: 'Learn how to use Office Scripts to filter an Excel table and get the visible range as an array of objects.'
+ms.date: 03/04/2021
+localization_priority: Normal
+---
+
+# Filter Excel table and get visible range as array of objects
+
+Shows how to use Office Scripts to filter an Excel table and get the visible range as an array of objects
+
+## Example scenario
+
+* Apply filter on a column of a table.
+* Extract visible range after filtering and extract data as array of objects.
+* Assemble an object with following JSON structure and return for a future action to use.
+
+### Sample JSON
+
+Each key represents a unique value of a table. Each array instance represents the row that is visible when the corresponding filter is applied.
+
+```json
+{
+  "Station-1": [{
+    "Station": "Station-1",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Debra Berger",
+    "Reason": ""
+  }, {
+    "Station": "Station-1",
+    "Shift": "Afternoon",
+    "Date": "27-Oct-20",
+    "Responsible": "Delia Dennis",
+    "Reason": ""
+  }, {
+    "Station": "Station-1",
+    "Shift": "Afternoon",
+    "Date": "28-Oct-20",
+    "Responsible": "Lidia Holloway",
+    "Reason": ""
+  }],
+  "Station-2": [{
+    "Station": "Station-2",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Gerhart Moller",
+    "Reason": ""
+  }, {
+    "Station": "Station-2",
+    "Shift": "Afternoon",
+    "Date": "28-Oct-20",
+    "Responsible": "Grady Archie",
+    "Reason": ""
+  }],
+  "Station-3": [{
+    "Station": "Station-3",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Isaiah Langer",
+    "Reason": ""
+  }]
+}
+```
+
+## Solution
+
+The solution has a single Office Script:
+
+* Performs transformation as described previously and extracts required data
+
+## Office Script
+
+The following script filters a table and gets the visible range.
+
+Download the sample file <a href="table-filter.xlsx">table-filter.xlsx</a> and use it with this script to try it out yourself!
+
+```ts
+function main(workbook: ExcelScript.Workbook): ReturnTemplate {
+  const table1 = workbook.getTable("Table1");
+  const keyColumnValues: string [] = table1.getColumnByName('Station').getRangeBetweenHeaderAndTotal().getValues().map(v => v[0] as string);
+  // const uniqueKeys= [...Array.from(new Set(keyColumnValues))];
+  const uniqueKeys = keyColumnValues.filter((v, i, a) => a.indexOf(v) === i);
```

Comment:
This map building code is super dense. It would greatly benefit from being split into several lines and having comments.

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)